openage/convert/processor/export/modpack_exporter.py

```diff
@@ -65,7 +65,7 @@ def export(modpack, args):
                 kwargs["palettes"] = args.palettes
 
             for request in cur_export_requests:
-                request.save(sourcedir, modpack_dir, **kwargs)
+                request.save(sourcedir, modpack_dir, args.compression_level, **kwargs)
```

Comment:
This needs to be in `kwargs` rather than in the expicit function call. The reason for this is that `SoundExportRequest` does not need the PNG compression level.

And the media export request's `save()` function must propagate the compression level to the texture's `save()` function, of course.

Reply:
Thanks for the pointers, I was wondering what is the use of kwargs here.

So compression level should apply only on `MediaType.GRAPHICS` and `MediaType.TERRAIN`?

Reply:
> So compression level should apply only on MediaType.GRAPHICS and MediaType.TERRAIN?

Correct

> Thanks for the pointers, I was wondering what is the use of kwargs here.

We are using `kwargs` here because not all media export requests require the same settings. It's a bit of a catch-all parameter. Once https://github.com/SFTtech/openage/issues/1311 is implemented, the could will hopefully be more understandable in that regard :)

Reply:
> We are using kwargs here because not all media export requests require the same settings. It's a bit of a catch-all parameter. Once #1311 is implemented, the could will hopefully be more understandable in that regard :)

With this explanation it is indeed clear, though it is confusing at first :)

> And the media export request's save() function must propagate the compression level to the texture's save() function, of course.

I'm a little lost at where to find particular export request `save()` function.

I guess it should be some of `graphic_export` used in https://github.com/SFTtech/openage/blob/master/openage/convert/processor/conversion/aoc/modpack_subprocessor.py#L124

But at that point I'm not sure where are these `*_export` defined and where should I look.

Reply:
They can all be found here: https://github.com/SFTtech/openage/blob/master/openage/convert/entity_object/export/media_export_request.py

Reply:
Thanks.

Just small question. why do we pass game_version into MediaType.GRAPHICS, if is not used here?

Reply:
Oh, we indeed do not have to pass it anymore. Before 3a9bc4f1706feead417428aac285f76461b7468e `Texture` required game_version to find the palette number. This requirement was removed, so putting it in `kwargs` for the GRAPHICS media type is a leftover code snippet. You can remove it.

Reply:
Resolved in 37ceb58fb3daf61c36feddad6a18e0da58ce8526
Please review it again.